src/TMxpMudlet.cpp

```diff
@@ -114,3 +114,55 @@ TLinkStore& TMxpMudlet::getLinkStore()
 {
     return mpHost->mpConsole->getLinkStore();
 }
+
+// Handle 'stacks' of attribute settings:
+void TMxpMudlet::setBold(bool bold)
+{
+    if (bold) {
+        boldCtr++;
+    } else if (boldCtr > 0) {
+        boldCtr--;
+    }
+}
+
+void TMxpMudlet::setItalic(bool italic)
+{
+    if (italic) {
+        italicCtr++;
+    } else if (italicCtr > 0) {
+        italicCtr--;
+    }
+}
+
+void TMxpMudlet::setUnderline(bool underline)
+{
+    if (underline) {
+        underlineCtr++;
+    } else if (underlineCtr > 0) {
+        underlineCtr--;
+    }
+}
+
+void TMxpMudlet::setStrikeOut(bool strikeOut)
+{
+    if (strikeOut) {
+        strikeOutCtr++;
+    } else if (strikeOutCtr > 0) {
+        strikeOutCtr--;
+    }
+}
+
+// reset text Properties (from open tags) at end of line
```

Comment:
Ok, I added a comment:
    // for the next two, we can assume both lists are usually empty (in case of properly
    // balanced MXP tags, and if not, they'll only contain very few entries. Thus it seems
    // sensible to check first, and then just remove all of the few nodes rather than
    // removing the whole list first and then create a new one.
 and wrote it that way since I think it is the most ressource friendly way to do it in the vast majority of cases. But I think you might prefer something like:
 
    if (!fgColors.isEmpty()) {
        fgColors = QList<QColor>();
    }
    
 Just drop me a note if I should go for that one. In principle, all of these are fine for me.

Reply:
This works ok!